### PR TITLE
remove support for transform arguments without partial in decorators

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
We used to advise users to pass transforms additional arguments positionally in transforms being used as decorators. Now, it's forbidden (they need to use `functools.partial`). It has been deprecated for 2 releases, so it's time to remove it.

**Description of the Change:**
Raise a `TransformError` if someone uses the old transform decorator syntax

**Benefits:**
Less support for old style of doing things, helps us keep our code cleaner

**Possible Drawbacks:**
It felt intuitive for me to turn the warning into an explicit error, but usually we just remove code outright when doing removals. This constructor doesn't have an intuitive way of just removing this functionality, so I feel like raises an error explicitly makes sense

[sc-51278]

### CODE REVIEW

File: doc/development/deprecations.rst
Comment (timmysilv) [inline doc/development/deprecations.rst:1]: I just moved it down and replaced "deprecated" with "removed", but git doesn't know any better

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -41,6 +41,10 @@
 
 <h3>Breaking changes 💔</h3>
 
+* Passing additional arguments to a transform that decorates a QNode must be done through the use
+  of `functools.partial`.
+  [()]()
```
Comment (timmysilv) [inline doc/releases/changelog-dev.md:46]:
```suggestion
  [(#5046)](https://github.com/PennyLaneAI/pennylane/pull/5046)
```

File: pennylane/transforms/core/transform_dispatcher.py [hunk 2]
```diff
@@ -127,28 +127,14 @@ def processing_fn(results):
         #
         # result = some_transform(*transform_args)(qnode)(*qnode_args)
 
-        warnings.warn(
+        raise TransformError(
```
Comment (rmoyard) [inline pennylane/transforms/core/transform_dispatcher.py:130]: Nice!

**Review by rmoyard — APPROVED** [review]
Thanks :100:

File: doc/development/deprecations.rst
Comment (trbromley) [inline doc/development/deprecations.rst:1]: This diff seems weird, but maybe just a quirk of how things were moved around?

**Review by trbromley — APPROVED** [review]
Thanks @timmysilv!

File: doc/development/deprecations.rst
Comment (timmysilv) [reply doc/development/deprecations.rst:1]: yep, commented earlier but it's now outdated

**Pull request merged**